src/docker/ContextManager.ts

```diff
@@ -80,16 +80,27 @@ export class DockerContextManager implements ContextManager, Disposable {
         // The file watchers are not strictly necessary; they serve to help the extension detect context switches
         // that are done in CLI. Worst case, a user would have to restart VSCode.
         try {
-            if (fse.existsSync(dockerConfigFile)) {
-                this.configFileWatcher = fs.watch(dockerConfigFile, async () => this.refresh());
+            if (fse.existsSync(path.join(dockerConfigFolder, dockerConfigFile))) {
+                this.configFileWatcher = workspace.createFileSystemWatcher(
+                    new RelativePattern(dockerConfigFolder, dockerConfigFile),
+                    true, // Don't expect file to be created (since we already verified its existence)
+                    false, // Do expect file to change
+                    true // Don't expect file to be deleted (are they uninstalling?!)

```

Comment:
Are you sure we want to ignore deletions? If Docker is uninstalled, we probably want to "refresh"...

Reply:
Yes, the latter is much better. My suggestion is then to keep the code as-is, but just massage the comment a bit, explaining why we do not need to watch for file deletion/Docker uninstall.